### PR TITLE
Adding proxy values to build args

### PR DESCRIPTION
**Pull request by donny-dont**
This makes it so the plugin implicitly adds proxy values as build args. It is done implicitly as the proxy values are contained within the agent and the information might not be known to the person creating the build.

Currently it checks to see if the values were set as build args and it will ignore the agent's values.

Also it sets the upper and lower case versions of the proxy values as I've found that this needs to happen.

@bradrydzewski @tboerger @Bugagazavr

### CODE REVIEW

File: plugin.go [hunk 1]
```diff
@@ -108,6 +108,11 @@ func (p Plugin) Exec() error {
 		fmt.Println("Registry credentials not provided. Guest mode enabled.")
 	}
 
+	// add proxy build args
+	addProxyBuildArgs(&p.Build)
+
+	fmt.Printf("Build arg length %d", len(p.Build.Args))
```
Comment (tboerger) [inline plugin.go]: Do you want to keep that debug info?

**Comment by tboerger** [comment]
LGTM

**Comment by tboerger** [comment]
Now it's really fine for me

**Comment by Bugagazavr** [comment]
LGTM

**Pull request merged**